torch/csrc/utils/python_arg_parser.h

```diff
@@ -389,9 +390,46 @@ inline std::vector<int64_t> PythonArgs::intlist(int i) {
   return intlistWithDefault(i, signature.params[i].default_intlist);
 }
 
+static bool is_symint_node(py::handle obj) {
+      auto static tp_symn = py::type::of<c10::SymbolicIntNode>();
+      return obj.get_type().equal(tp_symn);
+}
+
 inline std::vector<c10::SymInt> PythonArgs::symintlist(int i) {
-  auto intlist = intlistWithDefault(i, signature.params[i].default_intlist);
-  return c10::fmap(intlist, [](int64_t n) {return c10::SymInt(n); });
+
+  if (!args[i]) {
+    return c10::fmap(signature.params[i].default_intlist, [](int64_t di) {
+      return c10::SymInt(di);
+    });
+  }
+
+  const auto size1 = signature.params[i].size;
+  if (size1 > 0 && THPUtils_checkLong(args[i])) {
+    return std::vector<c10::SymInt>(size1, c10::SymInt(THPUtils_unpackIndex(args[i])));
+  }
```

Comment:
For symmetry, you also have to test if you got a singleton SymInt and convert that into a list as well